test/groovy/com/sap/piper/TemporaryCredentialsUtilsTest.groovy

```diff
@@ -0,0 +1,136 @@
+package com.sap.piper
+
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+import org.junit.rules.RuleChain
+import util.BasePiperTest
+import util.JenkinsCredentialsRule
+import util.JenkinsFileExistsRule
+import util.JenkinsLoggingRule
+import util.JenkinsReadFileRule
+import util.JenkinsReadYamlRule
+import util.JenkinsShellCallRule
+import util.JenkinsWriteFileRule
+import util.Rules
+
+import static org.junit.Assert.assertThat
+import static org.hamcrest.Matchers.*
+import static org.junit.Assert.assertTrue
+
+class TemporaryCredentialsUtilsTest extends BasePiperTest {
+    private ExpectedException thrown = ExpectedException.none()
+    private JenkinsCredentialsRule credentialsRule = new JenkinsCredentialsRule(this)
+    private JenkinsWriteFileRule writeFileRule = new JenkinsWriteFileRule(this)
+    private JenkinsFileExistsRule fileExistsRule = new JenkinsFileExistsRule(this, [])
+    private JenkinsReadFileRule readFileRule = new JenkinsReadFileRule(this, null)
+    private JenkinsReadYamlRule readYamlRule = new JenkinsReadYamlRule(this)
+    private JenkinsShellCallRule shellRule = new JenkinsShellCallRule(this)
+    private JenkinsLoggingRule loggingRule = new JenkinsLoggingRule(this)
+
+    def bodyExecuted
+    TemporaryCredentialsUtils credUtils
+
+    @Rule
+    public RuleChain ruleChain = Rules
+        .getCommonRules(this)
+        .around(thrown)
+        .around(readYamlRule)
+        .around(credentialsRule)
+        .around(loggingRule)
+        .around(writeFileRule)
+        .around(fileExistsRule)
+        .around(readFileRule)
+        .around(shellRule)
+
+    @Before
+    void init() {
+        bodyExecuted = false
+
+        credentialsRule.reset()
+            .withCredentials('erp-credentials', 'test_user', '********')
+            .withCredentials('testCred2', 'test_other', '**')
+
+        credUtils = new TemporaryCredentialsUtils(nullScript)
+    }
+
+    @Test
+    void credentialsFileWritten() {
+        def credential = [alias: 'ERP', credentialId: 'erp-credentials']
+        def directory = './'
+        def filename = 'credentials.json'
+        fileExistsRule.registerExistingFile('systems.yml')
+
+        credUtils.writeCredentials([credential], directory, filename )
+
+        assertThat(writeFileRule.files['credentials.json'], containsString('"alias":"ERP","username":"test_user","password":"********"'))
+    }
+
+    @Test
+    void credentialsFileNotWrittenWithEmptyList() {
+        def directory = './'
+        def filename = 'credentials.json'
+        fileExistsRule.registerExistingFile('systems.yml')
+
+        credUtils.writeCredentials([], directory, filename )
+
+        loggingRule.expect('Not writing any credentials.')
```

Comment:
Doesn't this mean that [the check](https://github.com/SAP/jenkins-library/pull/1894/files#diff-920223b0336ff97298b7005a29b996f1R20) doesn't work? That would imply you try to delete the non-existing file later.

From local testing, it seems `rm -f somefile` does not exit with an error if the somefile does not exist. So probably not a problem, but maybe still worth investigating.

Reply:
Bogus. Sorry for the noise. You are calling writeCredentials directly.